src/main/java/uk/gov/pay/ledger/agreement/state/AgreementState.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.pay.ledger.agreement.state;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import uk.gov.pay.ledger.event.model.SalientEventType;
+import uk.gov.service.payments.commons.model.agreement.AgreementStatus;
+
+import java.util.Map;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+public enum AgreementState {
```

Comment:
Do we need both `AgreeementState` and `AgreementStatus` enums as it's a 1:1 mapping? I'd say that resolving the status should go directly in `AgreementStatus` unless there's a reason it wouldn't work.

Reply:
I wondered about this.  I'm happy to kill off `AgreementState` as it's currently redundant.  If we ever end up with anything else that needs to go in an `AgreementState`, we can always resurrect it.